### PR TITLE
Add support for bridging NavSat

### LINKED ISSUE #218: NavSat Ignition Gazebo ROS Bridge Update
Hello, 

I am curious if anyone has been working on getting the NavSat/GPS parameter in Ignition Gazebo converted with the ign_ros bridge. Using Citadel I was able to get the NavSat Ignition Gazebo sensor message to work, but I didn't see the ROS message conversion available in the sensor list. Any tips on how I can get the GPS sensor message converted to ROS.

### PR DESCRIPTION
**Pull request by TyHowellWork**
# 🎉 New feature

Closes #218

## Summary
Adds two-way bridge for NavSat messages between ROS and IGN along with a demo launch file.

## Test it
Test IGN->ROS by running `roslaunch ros_ign_gazebo_demos navsat.launch`.
Test ROS->IGN by `roslaunch ros_ign_gazebo_demos navsat.launch` and `ign topic -t /navsat -e`
```
rostopic pub -r 1 /navsat sensor_msgs/NavSatFix "header:
  seq: 10
  stamp: {secs: 10, nsecs: 0}
  frame_id: 'back_to_ign'
status: {status: 0, service: 0}
latitude: 10.0
longitude: 10.0
altitude: 10.0
position_covariance: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
position_covariance_type: 0"
```
Even using the contributing guides I was unable to get codecheck or the tests running :(

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by chapulina — CHANGES_REQUESTED** [review]
Thanks for the PR. The implementation, test and example look great!

The only catch is that this branch is supported with both Citadel and Fortress, but the `NavSat` message and the `NavSat` sensor are only available from Fortress. So we have a couple of options:

* Add `ifdef`s so the `NavSat` is only present on `IGN_MSGS_VER >= 8`
* Or backport the `NavSat` message to `ign-msgs5` - while this is trivial, backporting the `NavSat` sensor itself isn't. It shouldn't hurt having the message there though.

In any case, we should add a note to `ros_ign_gazebo_demos/README.md` that the GNSS example only works from Fortress.

File: ros_ign_bridge/test/test_utils.h [hunk 1]
```diff
@@ -832,6 +833,42 @@ namespace testing
...
+  void compareTestMsg(const sensor_msgs::NavSatFix &_msg)
+  {
+    sensor_msgs::NavSatFix expected_msg;
+    createTestMsg(expected_msg);
+
+    compareTestMsg(_msg.header);
+    EXPECT_EQ(expected_msg.status, _msg.status);
+    EXPECT_FLOAT_EQ(expected_msg.latitude,           _msg.latitude);
+    EXPECT_FLOAT_EQ(expected_msg.longitude,          _msg.longitude);
+    EXPECT_FLOAT_EQ(expected_msg.altitude,           _msg.altitude);
```
Comment (ahcorde) [inline ros_ign_bridge/test/test_utils.h]:
```suggestion
    EXPECT_FLOAT_EQ(expected_msg.latitude, _msg.latitude);
    EXPECT_FLOAT_EQ(expected_msg.longitude, _msg.longitude);
    EXPECT_FLOAT_EQ(expected_msg.altitude, _msg.altitude);
```

File: ros_ign_bridge/test/test_utils.h [hunk 2]
```diff
@@ -832,6 +833,42 @@ namespace testing
...
+  {
+    std_msgs::Header header_msg;
+    createTestMsg(header_msg);
+
+    _msg.header                    = header_msg;
+    _msg.status.status             = sensor_msgs::NavSatStatus::STATUS_FIX;
+    _msg.latitude                  = 0.00;
+    _msg.longitude                 = 0.00;
+    _msg.altitude                  = 0.00;
+    _msg.position_covariance = {1, 2, 3, 4, 5, 6, 7, 8, 9};
```
Comment (ahcorde) [inline ros_ign_bridge/test/test_utils.h]:
```suggestion
    _msg.altitude                  = 0.00;
    _msg.position_covariance       = {1, 2, 3, 4, 5, 6, 7, 8, 9};
```

**Review by ahcorde — DISMISSED** [review]

**Comment by TyHowellWork** [comment]
@chapulina @ahcorde Thanks for the reviews I should have a fix :) by the end of the day.

**Comment by TyHowellWork** [comment]
ign-msgs5 [PR](https://github.com/ignitionrobotics/ign-msgs/pull/231) has been submitted that backports NavSat message only.

**Comment by chapulina** [comment]
> ign-msgs5 https://github.com/ignitionrobotics/ign-msgs/pull/231 has been submitted that backports NavSat message only.

Thanks! Here are the next steps:

- [x] Release `ign-msgs5` with the `NavSat` message: https://github.com/ignitionrobotics/ign-msgs/pull/232
- [x] Pull that into packages.ros.org with reprepro
- [ ] Run CI on this PR

**Review by chapulina — APPROVED** [review]
Ready! Thanks for the wait!

**Pull request merged**